stacks/api/passport/apikey/apikey.controller.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+  UseGuards
+} from "@nestjs/common";
+import {AuthGuard} from "@nestjs/passport";
+import {Schema} from "@spica-server/core/schema";
+import {ObjectId, OBJECT_ID} from "@spica-server/database";
+import * as uniqid from "uniqid";
+import {ActionGuard} from "../policy/action.guard";
+import {ApiKeyService} from "./apikey.service";
+import {ApiKey} from "./interface";
+
+@Controller("passport/apikey")
+export class ApiKeyController {
+  constructor(private aks: ApiKeyService) {}
+
+  @Get()
+  @UseGuards(AuthGuard(), ActionGuard("passport:apikey:index"))
+  find() {
+    return this.aks.find();
+  }
+
+  @Get(":id")
+  @UseGuards(AuthGuard(), ActionGuard("passport:apikey:show"))
+  findOne(@Param("id", OBJECT_ID) id: ObjectId) {
+    return this.aks.findOne({_id: id}).then(r => {
+      if (!r) {
+        throw new NotFoundException();
+      }
+      return r;
+    });
+  }
+
+  @Post()
+  @UseGuards(AuthGuard(), ActionGuard("passport:apikey:add"))
+  insertOne(@Body(Schema.validate("http://spica.internal/passport/apikey")) apiKey: ApiKey) {
+    delete apiKey._id;
+    apiKey.key = uniqid();
+    return this.aks.insertOne(apiKey).then(r => {
+      apiKey._id = r;
+      return apiKey;
+    });
+  }
+
+  @Post(":id")
```

Comment:
Is there any specific reason to use **Post** for update and insert actions?

Reply:
This could be helpful: https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html